test/index.html

```diff
@@ -0,0 +1,129 @@
+  limitations under the License.
+-->
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>SW Toolbox Tests</title>
+  <link href="/node_modules/mocha/mocha.css" rel="stylesheet" />
+
+  <!--
+    iframes are used to manage service worker scoping.
+    This will hide them and stop the page from jumping around
+  -->
+  <style>
+    iframe {
+      width: 0;
+      height: 0;
+    }
+  </style>
+</head>
+<body>
+  <div id="mocha"></div>
+
+  <script src="/node_modules/chai/chai.js"></script>
+  <script src="/node_modules/mocha/mocha.js"></script>
+
+  <!-- Helper functions will be under window.testHelper -->
+  <script src="/test/libs/helper-functions.js"></script>
+
+  <!--
+    Timeout is extended to ensure tests for max-cache-age
+    have enough time to complete
+  -->
+  <script>mocha.setup({
+    ui: 'bdd',
+    timeout: 5000
+  })</script>
+
+  <!-- In browser test scripts should be added to the page here-->
+  <script src="/test/browser-tests/precache/precache.js"></script>
+
+  <script>
+    (function() {
+      // To automate the tests with selenium, we need a way
+      // to inform selenium when the tests are completed and
+      // pass results back.
+      // This is done by waiting for window.swtoolbox to be defined.
+      // This is done in publishTestResults();
+      var passedTests = [];
+      var failedTests = [];
+
+      // This is a helper method that simplifies the test
+      // results to an object that can be JSON.string-ified.
+      function getFriendlyTestResult(testResult) {
+        return {
+          title: testResult.title,
+          state: testResult.state
+        };
+      };
+
+      // Once this is called, automated tests should be
+      // able to read sw.toolbox and get test results.
+      function publishTestResults() {
+        window.swtoolbox = {
+          testResults: {
+            passed: passedTests,
+            failed: failedTests
+          }
+        };
+      }
+
+      // This make browsers without a service worker pass tests by
+      // bypassing the tests altogether.
+      // This is desirable to allow travis to run tests in all browsers
+      // regardless of support or not and perform tests when the browser
+      // starts to support service workers.
+      if (!('serviceWorker' in navigator)) {
+        publishTestResults();
+        return;
+      }
+
+      window.chai.should();
+
+      // We unregister all service workers, clear all caches and remove
+      // All registered iframes
+      beforeEach(function(done) {
+        testHelper.cleanState()
+        .then(() => done(), done);
+      });
+
+      // Clean up after the final test has finished
+      after(function(done) {
+        testHelper.cleanState()
+        .then(() => done(), done);
+      });
+
+      var runResults = mocha.run();
+
+      // If there are no tests we should define window.swtoolbox
+      // so the automated test suite will know the test has ended
+      if (runResults.total === 0) {
```

Comment:
runResults.total (i.e. the number of tests to be run) is set my mocha - but the results of pass and fail can't be known until the tests have actually been run.

The ONLY reason for the publishTestResults is to ensure that the selenium tests end as soon as possible. This was likely added to account for no tests in browsers that don't support service worker. This is still a handy fail safe, but agreed, serves little purpose for our tests. Will remove.